tools/build_script_generator/module.lb

```diff
@@ -157,6 +152,10 @@ def post_build(env):
         "gitignore": gitignore,
         "elf_files": elf_files,
     }
+    # prepare custom path
+    openocd_cfg = env.get(":build:openocd.cfg", "")
```

Comment:
I think we should warn user or even generate an error if this option is empty.

Reply:
Good idea, I've implemented this as a warning displaying part of the description again:
```
 $ lbuild build
[WARNING] user.modm.board.blue-pill: You need to provide the programmer via a custom OpenOCD config!

## Programming

Since the board doesn't have a programmer on-board, you need to use your
own and *specify* which one you're using in a custom `openocd.cfg` file:

    # Replace this with your custom programmer
    source [find interface/stlink-v2.cfg]

    # If you use the clone CKS32F103C8T6 chip you need to overwrite this ID
    #set CPUTAPID 0x2ba01477

Then include this file in your build options like so:

    <library>
      <extends>modm:blue-pill</extends>
      <options>
        <option name="modm:build:openocd.cfg">openocd.cfg</option>
      </options>
    </library>
```